tensorflow_graphics/datasets/shapenet/shapenet.py

```diff
@@ -0,0 +1,183 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Lint as: python3
+"""Shapenet Core dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import csv
+import json
+import os
+import textwrap
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+from tensorflow_datasets import features as tfds_features
+from tensorflow_graphics.datasets import features as tfg_features
+
+_CITATION = """
+@techreport{shapenet2015,
+  title       = {{ShapeNet: An Information-Rich 3D Model Repository}},
+  author      = {Chang, Angel X. and Funkhouser, Thomas and Guibas, Leonidas and Hanrahan, Pat and Huang, Qixing and Li, Zimo and Savarese, Silvio and Savva, Manolis and Song, Shuran and Su, Hao and Xiao, Jianxiong and Yi, Li and Yu, Fisher},
+  number      = {arXiv:1512.03012 [cs.GR]},
+  institution = {Stanford University --- Princeton University --- Toyota Technological Institute at Chicago},
+  year        = {2015}
+}
+"""
+
+_DESCRIPTION = """
+ShapeNetCore is a densely annotated subset of ShapeNet covering 55 common object
+categories with ~51,300 unique 3D models. Each model in ShapeNetCore is linked
+to an appropriate synset in WordNet (version 3.0).
+
+The synsets will be extracted from the taxonomy.json file in the ShapeNetCore.v2.zip
+archive and the splits from http://shapenet.cs.stanford.edu/shapenet/obj-zip/SHREC16/all.csv
+"""
+
+_TAXONOMY_FILE_NAME = 'taxonomy.json'
+
+_MODEL_SUBPATH = os.path.join('models', 'model_normalized.obj')
+
+_SPLIT_FILE_URL = \
+    'http://shapenet.cs.stanford.edu/shapenet/obj-zip/SHREC16/all.csv'
+
+_CHECKSUMS_DIR = os.path.normpath(
+    os.path.join(os.path.dirname(__file__), 'checksums/'))
+
+
+class Shapenet(tfds.core.GeneratorBasedBuilder):
+  """ShapeNetCore V2.
+
+  Example usage of the dataset:
+
+  import tensorflow_datasets as tfds
+  from tensorflow_graphics.datasets.shapenet import Shapenet
+
+  data_set = Shapenet.load(
+      split='train',
+      download_and_prepare_kwargs={
+          'download_config':
+              tfds.download.DownloadConfig(manual_dir='~/shapenet_base')
+      })
+
+  for example in data_set.take(1):
+    trimesh, label, model_id = example['trimesh'], example['label'],
+                               example['model_id']
+
+  """
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  @staticmethod
+  def load(*args, **kwargs):
+    return tfds.load('shapenet', *args, **kwargs)
+
+  MANUAL_DOWNLOAD_INSTRUCTIONS = textwrap.dedent("""\
+  manual_dir should contain the extracted ShapeNetCore.v2.zip archive.
+  You need to register on https://shapenet.org/download/shapenetcore in order
+  to get the link to download the dataset.
+  """)
+
```

Comment:
Having tfds handle download/extraction is really convenient, and the link is a public URL. Is it possible to check `os.environ["SHAPENET_URL"]` or similar if `manual_dir` isn't provided? Checksums might be tricky to handle without exposing the URL..